### PR TITLE
Interactive base world + some fixes

### PR DESCRIPTION
**Pull request by jaseweston**
- get_contexts seemed to have missing param sometimes, sometimes not, breaking stuff (change by spencer i believe)

- issue: seems a few of the contexts had an extra '\n' at the end. (This might confuse the model about which turn is next, blended_self_talk had this problem in interactive too, I pushed separately.)

### CODE REVIEW

File: parlai/tasks/convai2/worlds.py [hunk 1]
```diff
@@ -44,8 +43,8 @@ def load_personas(opt):
                     )
                 if t.startswith('your persona:'):
                     a2_persona += t + '\n'
-            personas.add(a1_persona)
-            personas.add(a2_persona)
+            personas.add(a1_persona.rstrip('\n'))
```
Comment (emilydinan) [inline parlai/tasks/convai2/worlds.py]: why do we strip the new line here? is it becuase there is an extra newline at the end? maybe we shouldj ust edit the way we grab the personas. something like 
```
a1_persona = []
...
    a1_persona.append(t.replace("partner's persona:", "your persona:")
...
personas.add('\n'.join(a1_persona)

```

File: parlai/tasks/interactive/worlds.py [hunk 2]
```diff
@@ -16,4 +18,57 @@ class InteractiveWorld(DialogPartnerWorld):
...
+        """
+        pass
+
+    def get_contexts(self):
+        """
+        Override to return a pair of contexts with which to seed the episode.
+
+        This function will be called before the first turn of every episode.
+        """
+        return ['__SILENCE__', '']
```
Comment (emilydinan) [inline parlai/tasks/interactive/worlds.py]: do we want this to be the default? not all datasets have a silence token?

File: parlai/tasks/interactive/worlds.py [hunk 3]
```diff
@@ -16,4 +18,57 @@ class InteractiveWorld(DialogPartnerWorld):
...
+            self.p1, self.p2 = self.get_contexts()
+
+        acts = self.acts
+        agents = self.agents
+        if self.turn_cnt == 0:
+            # add the context on to the first message to agent 0
+            act = {}
+            act['text'] = self.p1
+            act['episode_done'] = False
+            act['id'] = 'context'
```
Comment (emilydinan) [inline parlai/tasks/interactive/worlds.py]: do we always want to add a contexT? i'm not sure about this 

cc @stephenroller

File: parlai/tasks/light_dialog/worlds.py [hunk 4]
```diff
@@ -72,48 +66,11 @@ def get_new_personas(self):
...
-        self.cnt += 1
-
-        if act['episode_done']:
-            print("CHAT DONE ")
-            print("\n... preparing new chat... \n")
-            self.cnt = 0
-
 
-class InteractiveSelfchatWorld(SelfChatBaseWorld):
+class SelfChatWorld(SelfChatBaseWorld):
```
Comment (emilydinan) [inline parlai/tasks/light_dialog/worlds.py]: missed this one -- thanks!

File: parlai/tasks/convai2/worlds.py (hunk 1 above)
Comment (jaseweston) [reply parlai/tasks/convai2/worlds.py]: sure, that would do the same thing

File: parlai/tasks/interactive/worlds.py (hunk 2 above)
Comment (jaseweston) [reply parlai/tasks/interactive/worlds.py]: yes, i can remove this.

File: parlai/tasks/interactive/worlds.py (hunk 3 above)
Comment (jaseweston) [reply parlai/tasks/interactive/worlds.py]: hmm, i guess i could just add if it isn't an empty string..

File: parlai/tasks/convai2/worlds.py (hunk 1 above)
Comment (jaseweston) [reply parlai/tasks/convai2/worlds.py]: (but yes, the extra newline was a bug)

File: parlai/tasks/interactive/worlds.py (hunk 3 above)
Comment (stephenroller) [reply parlai/tasks/interactive/worlds.py]: Does it make more sense to have get_context return something other than an empty string for that? E.g. it returns None instead?

This is fine i think, but maybe the semantics could be improved.

File: parlai/tasks/convai2/worlds.py [hunk 5]
```diff
@@ -4,15 +4,12 @@
...
-from copy import deepcopy
-
 from parlai.core.worlds import create_task
-from parlai.core.worlds import DialogPartnerWorld, validate
 from parlai.agents.fixed_response.fixed_response import FixedResponseAgent
 from parlai.tasks.self_chat.worlds import SelfChatWorld as SelfChatBaseWorld
+from parlai.tasks.interactive.worlds import InteractiveWorld as InteractiveBaseWorld
 
 import random
-from typing import List
```
Comment (stephenroller) [inline parlai/tasks/convai2/worlds.py]: lol oh no, even less typing?!

**Pull request merged**